### PR TITLE
[FS-1061]Add server delete API

### PR DESCRIPTION
**Pull request by Alva8756**
Add `ServerDelete` API.

Tested with `mctl` in `sandbox`.

### CODE REVIEW

**Review by DoctorVin — CHANGES_REQUESTED** [review]
As much as I loathe the client tests, I spent a bunch of time working on them last night and they should both build and test cleanly. Moreover, the client tests should also run on a simple invocation of `go test` now (instead of being gated by a `testtools` build tag.

You should add a test for this new API there.

**Comment by Alva8756** [comment]
> As much as I loathe the client tests, I spent a bunch of time working on them last night and they should both build and test cleanly. Moreover, the client tests should also run on a simple invocation of `go test` now (instead of being gated by a `testtools` build tag.
> 
> You should add a test for this new API there.

Thanks for helping make the `client_test.go` work again! I added some tests for `serverEnroll` and `serverDelete`. PTAL when you have time. Thanks!

**Review by DoctorVin — CHANGES_REQUESTED** [review]
Some minor changes in the tests, please.

File: pkg/api/v1/client/client_test.go [hunk 1]
```diff
@@ -72,6 +72,7 @@ func newTester(t *testing.T) (*integrationTester, finalizer) {
 		server.WithStore(repository),
 		server.WithFleetDBClient(fleetDBClient),
 		server.WithStreamBroker(stream),
+		server.EnableServerEnroll(true),
```
[the remark this answers is not in the document]
Comment (Alva8756) [reply pkg/api/v1/client/client_test.go:75]: Nice catch! Here is the link for the ticket: https://packet.atlassian.net/browse/FS-1066

File: pkg/api/v1/types/types.go [hunk 2]
```diff
@@ -46,6 +46,14 @@ type AddServerParams struct {
 	Password string `json:"pwd"`
 }
 
+func (asp *AddServerParams) MustJSON() []byte {
```
Comment (DoctorVin) [inline pkg/api/v1/types/types.go:49]: 🚀

**Review by DoctorVin — APPROVED** [review]
:shipit:

**Pull request merged**